packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -41,9 +43,19 @@ export interface ViewCustomTimings {
 export const THROTTLE_VIEW_UPDATE_PERIOD = 3000
 export const SESSION_KEEP_ALIVE_INTERVAL = 5 * ONE_MINUTE
 
-export function trackViews(location: Location, lifeCycle: LifeCycle) {
+export function trackViews(location: Location, lifeCycle: LifeCycle, newLocationListener?: NewLocationListener) {
   const startOrigin = 0
-  const initialView = newView(lifeCycle, location, ViewLoadingType.INITIAL_LOAD, document.referrer, startOrigin)
+  const onNewLocation: NewLocationListener = newLocationListener || (() => undefined)
+
+  const { viewName } = onNewLocation(location) || {}
```

Comment:
You could also use the `?.` operator like this (this way, you don't even have to set a default value for the parameter):
```ts
const viewName = onNewLocation?.(location)?.viewName
```